### PR TITLE
Allow regular expression patterns in jobs

### LINKED ISSUE #10: Make it easier to match wildcards at a single level
`x.y.*` matches both `x.y.z` and `x.y.z.w`. This is fine/desirable, but we also need to be able to say 'only match `x.y.z`', since something like `myapp.member.created` may be handled differently from `myapp.member.subresource.created`.

### PR DESCRIPTION
**Pull request by DeeTheDev**
Fixes #10

- Allow regular expression job patterns to match more specific events (check issues for more information)

### CODE REVIEW

File: lib/amigo.rb [hunk 1]
```diff
@@ -57,6 +57,10 @@
 # The 'on' pattern can be 'myapp.customer.*' to match all customer events for example,
 # or '*' to match all events. The rules of matching follow File.fnmatch.
 #
+# The 'on' pattern accepts regular expressions '/myapp\.customer\.created&/' to only match specific
```
Comment (rgalanakis) [inline lib/amigo.rb:60]: `The 'on' pattern also accepts regular expressions, like /^myapp\.customer\.[a-z]+$/, to control the matching rules more closely than `File.fnmatch` can provide.`

File: lib/amigo/router.rb [hunk 2]
```diff
@@ -11,7 +11,13 @@ class Router
     def perform(event_json)
       event_name = event_json["name"]
       matches = Amigo.registered_event_jobs.
-        select { |job| File.fnmatch(job.pattern, event_name, File::FNM_EXTGLOB) }
+        select do |job|
+        if job.pattern.instance_of?(Regexp)
```
Comment (rgalanakis) [inline lib/amigo/router.rb:15]:
```ruby
if job.pattern.respond_to?(:match)
   job.pattern.match(event_name)
else
  File.fnmatch(job.pattern, event_name, File::FNM_EXTGLOB)
end
```

Comment (DeeTheDev) [reply lib/amigo/router.rb:15]: @rgalanakis `.responds_to?(:match)` also matches strings, so this isn't a good solution to check if it's a regular expression. Got any other ideas?

Comment (rgalanakis) [reply lib/amigo/router.rb:15]: Ah, ok. Use `is_a?(Regexp)`

https://stackoverflow.com/questions/3893278/ruby-kind-of-vs-instance-of-vs-is-a

Comment (DeeTheDev) [reply lib/amigo/router.rb]: @rgalanakis Great, done. Now I know the differences of those methods. 👍🏼

**Pull request merged**